contrib/upgrade.sh

```diff
@@ -13,41 +11,83 @@ set -euo pipefail
 #   upgrade.sh 1.15.10
 #   giteahome=/opt/gitea giteaconf=$giteahome/app.ini upgrade.sh
 
+while true; do
+  case "$1" in
+    -v | --version ) ver="$2"; shift 2 ;;
+    -y | --yes ) no_confirm="yes"; shift ;;
+    --ignore-gpg) ignore_gpg="yes"; shift ;;
+    -- ) shift; break ;;
+    * ) break ;;
+  esac
+done
+
+set -euo pipefail
+
+
+function require {
+  for exe in "$@"; do
+    command -v "$exe" &>/dev/null || (echo "missing dependency '$exe'"; exit 1)
+  done
+}
+
+
+require curl xz sha256sum gpg
+
+if [[ -f /etc/os-release ]]; then
+  os_release=`cat /etc/os-release`
+
+  if [[ "$os_release" =~ "OpenWrt" ]]; then
+    sudocmd="su"
+    service_start="/etc/init.d/gitea start"
+    service_stop="/etc/init.d/gitea stop"
+    service_status="/etc/init.d/gitea status"
+  else
+    require systemctl
+  fi
+fi
+
+
 # apply variables from environment
 : "${giteabin:="/usr/local/bin/gitea"}"
 : "${giteahome:="/var/lib/gitea"}"
 : "${giteaconf:="/etc/gitea/app.ini"}"
 : "${giteauser:="git"}"
 : "${sudocmd:="sudo"}"
 : "${arch:="linux-amd64"}"
+: "${service_start:="$sudocmd systemctl start gitea"}"
+: "${service_stop:="$sudocmd systemctl stop gitea"}"
+: "${service_status:="$sudocmd systemctl status gitea"}"
 : "${backupopts:=""}" # see `gitea dump --help` for available options
 
-function giteacmd {
-  "$sudocmd" --user "$giteauser" "$giteabin" --config "$giteaconf" --work-path "$giteahome" "$@"
-}
 
-function require {
-  for exe in "$@"; do
-    command -v "$exe" &>/dev/null || (echo "missing dependency '$exe'"; exit 1)
-  done
+function giteacmd {
+  if [[ $sudocmd = "su" ]]; then
+    "$sudocmd" - "$giteauser" -c "$giteabin --config $giteaconf --work-path $giteahome $@"
+  else
+    "$sudocmd" --user "$giteauser" "$giteabin" --config "$giteaconf" --work-path "$giteahome" "$@"
+  fi
 }
-require systemctl curl xz sha256sum gpg "$sudocmd"
 
 # select version to install
-if [[ -z "${1:-}" ]]; then
+if [[ -z "${ver:-}" ]]; then
   require jq
   giteaversion=$(curl --connect-timeout 10 -sL https://dl.gitea.io/gitea/version.json | jq -r .latest.version)
 else
   giteaversion="$1"
```

Comment:
Shouldn't this be:


```suggestion
  giteaversion="$ver"
```

Perhaps you should just charge the variable to giteaversion and just drop this line?